pkl-gradle/src/main/java/org/pkl/gradle/task/BasePklTask.java

```diff
@@ -231,4 +241,17 @@ protected List<Pattern> patternsFromStrings(List<String> patterns) {
     @Nullable T value = provider.getOrNull();
     return value == null ? null : f.apply(value);
   }
+
+  protected Map<String, PklEvaluatorSettings.ExternalReader> parseExternalReaders(
+      Provider<Map<String, String>> provider) {
+    return provider.getOrElse(Map.of()).entrySet().stream()
+        .collect(
+            Collectors.toMap(
+                Map.Entry::getKey,
+                (it) -> {
+                  var cmd = StringsKt.shlex(it.getValue());
+                  return new PklEvaluatorSettings.ExternalReader(
+                      cmd.getFirst(), cmd.subList(1, cmd.size()));
```

Comment:
If the input value string is empty this throws an exception. Is there a better way to surface this validation of invalid input? The CLI does this:
```bash
> pkl eval pkl:base --external-resource-reader x=''
Usage: pkl eval [<options>] <modules>...

Error: invalid value for --external-resource-reader: List is empty.
```

Reply:
You can validate a property by throwing here (e.g. by throwing a `GradleException`).